src/corecontext.h

```diff
@@ -0,0 +1,28 @@
+// As shown here: https://github.com/mortennobel/QtOpenGL3.2Core
+
+#ifndef DISPLAZ_CORECONTEXT_H
+#define DISPLAZ_CORECONTEXT_H
+
+#include <QGLContext>
+#include <QGLFormat>
+
+#if defined(Q_OS_MAC)
+#if QT_VERSION < 0x050000 && QT_VERSION >= 0x040800 // if less than Qt 5.0.0
+void* select_3_2_mac_visual(GDHandle handle, int depthBufferSize, bool multiSample=false);
+#endif
+#endif
+
+struct CoreContext : public QGLContext {
+    CoreContext(const QGLFormat& format, QPaintDevice* device) : QGLContext(format,device) {}
+    CoreContext(const QGLFormat& format) : QGLContext(format) {}
+
+#if defined(Q_OS_MAC)
+#if QT_VERSION < 0x050000 && QT_VERSION >= 0x040800 // if less than Qt 5.0.0
+    virtual void* chooseMacVisual(GDHandle handle) {
+        return select_3_2_mac_visual(handle, this->format().depthBufferSize()); //TODO: check if we want to enable multiSampling
+    }
+#endif
+#endif
+};
+
+#endif //DISPLAZ_CORECONTEXT_H
```

Comment:
We can go with Qt5 on OS X. There is no need to keep Qt4 support around. There are official binaries for both versions, but Qt5 is mature enough on OS X to make it our requirement - it also fixes a number of bugs, which we don't have to deal with or maintain obscure workarounds for.